### PR TITLE
Upgrade granulate-utils and add a test for resolve_proc_root_links()

### PR DESCRIPTION
**Pull request by Jongy**
Upgrade granulate-utils, this includes 2 new commits:
1. Some changes to `resolve_proc_root_links()` - we didn't have a test that triggers the underlying logic in this function, so I hacked one now that tests it to full extent. CC @ty93 
2. `elf.py` moved to granulate-utils, use the copy from there.

### CODE REVIEW

**Comment by Jongy** [comment]
Forgot to mention - I ran the new test with the old code (not the updated granulate-utils) and obv it fails.

File: tests/test_java.py [hunk 1]
```diff
@@ -416,3 +416,47 @@ def test_java_noexec_dirs(
...
+    treatment is required for /proc/pid/root paths), and that profiling works eventually.
+    This basicaly tests the function resolve_proc_root_links().
+    """
+    caplog.set_level(logging.DEBUG)
+
+    # build this structure
+    # /run/final_tmp
+    # /run/step2 -> final_tmp
+    # /run/step1 -> step2
+    # /run/tmp -> /run/step1
```
Comment (adi-benz) [inline tests/test_java.py:440]: `/run/tmpy`

File: tests/test_java.py [hunk 2]
```diff
@@ -416,3 +416,47 @@ def test_java_noexec_dirs(
...
+    # /tmp -> /run/tmp
+    application_docker_container.exec_run(
+        [
+            "sh",
+            "-c",
+            "mkdir -p /run/final_tmp && "
+            "ln -s final_tmp /run/step2 && "
```
Comment (adi-benz) [inline tests/test_java.py:447]:
```suggestion
            "ln -s final_tmp /run/step2 && "  # Test relative path
```

File: tests/test_java.py [hunk 3]
```diff
@@ -416,3 +416,47 @@ def test_java_noexec_dirs(
...
+            "ln -s step2 /run/step1 && "
+            "ln -s /run/step1 /run/tmpy && "
```
Comment (adi-benz) [inline tests/test_java.py:449]:
```suggestion
            "ln -s /run/step1 /run/tmpy && "  # Test absolute path
```

**Review by adi-benz — CHANGES_REQUESTED** [review]

**Review by adi-benz — APPROVED** [review]

**Pull request merged**